pkg/nmi/iptables/iptables.go

```diff
@@ -89,9 +89,28 @@ func ensureCustomChain(ipt *iptables.IPTables, destIP, destPort, targetip, targe
 
 		For this reason we check if the length of rules is 3. If not 3, then we flush and create chain again.
 	*/
-	if len(rules) == 3 {
+
+	expectedRules := map[string]struct{}{
+		"-N aad-metadata": {},
+		"-A aad-metadata ! -s 127.0.0.1/32 -d 169.254.169.254/32 -p tcp -m tcp --dport 80 -j DNAT --to-destination 127.0.0.1:2579": {},
```

Comment:
`2579` is the default NMI port but it can be changed via `--nmi-port` flag. Let's account that scenario as well.